sharktank/sharktank/models/wan/tools/export_all.py

```
@@ -0,0 +1,235 @@
+            artifacts_path, f"{model_name}_transformer_{dtype}.irpa"
+        )
+        export_wan_transformer_from_hugging_face(
+            repo_id=wan_repo,
+            mlir_output_path=mlir_path,
+            parameters_output_path=weights_path,
+            batch_sizes=[batch_size],
+            height=height,
+            width=width,
+            num_frames=num_frames,
+        )
+        print(f"✅ Saved Transformer MLIR to: {mlir_path}")
+        if return_paths:
+            return mlir_path, weights_path
+        return
+
+    else:
+        raise NotImplementedError()
+
+
+# --- Main Execution Block ---
+
+
+def main():
+    """Parses command-line arguments and runs the export process."""
+    parser = argparse.ArgumentParser(description="Export WANI2V Model Components")
+    parser.add_argument(
+        "--wan_repo",
+        type=str,
+        default="wan-AI/Wan2.1-T2V-14B",
+        help="Hugging Face model repository (required for transformer).",
+    )
+    parser.add_argument("--width", type=int, default=512)
+    parser.add_argument("--height", type=int, default=512)
+    parser.add_argument("--num_frames", type=int, default=81)
+    parser.add_argument("--batch_size", type=int, default=1)
+    parser.add_argument(
+        "--export",
+        type=str,
+        default="all",
+        help="Component(s) to export. Comma-separated: 't5', 'clip', 'vae', 'transformer', or 'all'.",
+    )
+    args = parser.parse_args()
+    logging.basicConfig(level=logging.INFO)
+
+    all_components = ["clip", "t5", "vae", "transformer"]
+
+    if args.export.lower() == "all":
+        components_to_export = all_components
+    else:
+        components_to_export = [
+            c.strip() for c in args.export.split(",") if c.strip() in all_components
+        ]
+        if not components_to_export:
+            print(
+                f"No valid components specified. Please choose from: {', '.join(all_components)}"
+            )
+            return
```

This will silently succeed when it should fail.